### DISCUSSION TITLE
Display the next character in the centre of the xpad

### DISCUSSION THREAD

**Discussion by kjoetom** [8VIM/8VIM#379] [Ideas]
Would it be possible to always show in the circle the letter or character that would be written if you returned to the circle at that exact moment?

The advantages from my point of view would be:
- Immediate visual feedback as you type
- Even the letters of the hidden layer become visible before you write them
- Reduced learning curve for the new input method
- Avoidance of errors or the ability to correct before finishing typing (which could be particularly helpful for more complex movement patterns).

Displaying the next letter in the centre would have the advantage that this location would always be clearly visible during typing movements.

If this has already been discussed, I apologise (I couldn't find a discussion on this topic)

kjoe

**Comment by supercheetah** [comment]
This would be a solution to issue #420. I had the idea of putting the current row up near the top of the keyboard with the current letter highlighted. I put a gif animation example of that in the issue.

**Comment by kjoetom** [comment]
I'm not sure what the current opinion is here regarding word prediction. I believe the lack of word prediction is one of the reasons why 8vim hasn't been more widely adopted. Unlike keyboards with small keys that are often missed, 8vim ensures that the first typed letters are always accurate, making predictions easier and more precise.

Why I bring this up here? Because the next letter and word prediction could be effectively combined in a "Suggestions Bar". This bar could be optional. Additionally, the display of the next letter could also be optional and configurable, allowing it to appear on either the left or right side based on whether the user is right-handed or left-handed.

What do you all think?
